### PR TITLE
Allow "friendsofsymfony/rest-bundle:^3.0"

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

Allow "friendsofsymfony/rest-bundle:^3.0".

I am targeting this branch, because these changes respect BC.

## Changelog

```markdown
### Added
- support for "friendsofsymfony/rest-bundle:^3.0"
### Change
- support for deprecated "rest" routing type in favor for xml
```

### CODE REVIEW

File: src/BasketBundle/Controller/Api/BasketController.php [hunk 1]
```diff
@@ -14,8 +14,7 @@
 namespace Sonata\BasketBundle\Controller\Api;
 
 use FOS\RestBundle\Context\Context;
-use FOS\RestBundle\Controller\Annotations\QueryParam;
-use FOS\RestBundle\Controller\Annotations\View;
+use FOS\RestBundle\Controller\Annotations as Rest;
 use FOS\RestBundle\Request\ParamFetcherInterface;
 use FOS\RestBundle\View\View as FOSRestView;
```
Comment (phansys) [inline src/BasketBundle/Controller/Api/BasketController.php:19]: The alias is not required anymore.

File: src/BasketBundle/Controller/Api/BasketController.php [hunk 2]
```diff
@@ -134,9 +134,9 @@ public function getBasketsAction(ParamFetcherInterface $paramFetcher)
      *  }
      * )
      *
-     * @View(serializerGroups={"sonata_api_read"}, serializerEnableMaxDepthChecks=true)
+     * @Rest\View(serializerGroups={"sonata_api_read"}, serializerEnableMaxDepthChecks=true)
      *
-     * @param $id
+     * @param int $id Basket identifier
```
Comment (phansys) [inline src/BasketBundle/Controller/Api/BasketController.php:139]: Is the model forcing the basket to allow only integer ids?

Comment (wbloszyk) [reply src/BasketBundle/Controller/Api/BasketController.php:139]: Model allow for `mixed`. I use `int` to be unequivocal with `@ApiDoc`

Comment (phansys) [reply src/BasketBundle/Controller/Api/BasketController.php:139]: There is an issue with the `@ApiDoc` annotation then. I think we should not propagate the same problem here.
I understand you're trying to be consistent, but in this case the reference type does not seem to be the right choice.

Comment (wbloszyk) [reply src/BasketBundle/Controller/Api/BasketController.php:139]: I reverted it (docblock type hints). Lets do it in seperate PR.

**Comment by wbloszyk** [comment]
@sonata-project/contributors Can you check it?

**Comment by wbloszyk** [comment]
I will add it at the end, after add support for nelmio/api-doc-bundle v3

**Comment by wbloszyk** [comment]
@OskarStark Can you look at DOCtor-RTS? You make somethig simmilar ([to this](https://github.com/sonata-project/ecommerce/pull/701)), where nelmio/api-doc work with `.yml`.

**Pull request merged**